Cozo server says it's binding to port 9070 but actually binds to port 3000

```
~/workspace $ cozo-bin server -e sqlite -p my.db 
[2023-12-29T16:18:59Z INFO  cozo_bin::server] Starting Cozo (sqlite-backed) API at http://127.0.0.1:9070
```

```
~/workspace $ sudo netstat -lpn | grep cozo
(standard input):tcp        0      0 127.0.0.1:3000          0.0.0.0:*               LISTEN      230/cozo-bin  
```

Note: I'd be happy to sign the contributor agreement and submit a PR for this, but I'm having trouble getting the project to build. It fails building cozorocks (and I am only using sqlite but not sure that matters) Any hints?
This is what the first errors look like
```
warning: /home/v/workspace/cozo/target/debug/build/cozorocks-4c4da2415a34542d/out/cxxbridge/sources/cozorocks/src/bridge/mod.rs.cc:875:58: error: ‘kMergeOperandThresholdExceeded’ is not a member of ‘StatusSubCode’ {aka ‘rocksdb::Status::SubCode’}
warning:   875 | static_assert(static_cast<::std::uint8_t>(StatusSubCode::kMergeOperandThresholdExceeded) == 16, "disagrees with the value in #[cxx::bridge]");
warning:       |                                                          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
warning: /home/v/workspace/cozo/target/debug/build/cozorocks-4c4da2415a34542d/out/cxxbridge/sources/cozorocks/src/bridge/mod.rs.cc:876:71: error: static assertion failed: disagrees with the value in #[cxx::bridge]
warning:   876 | static_assert(static_cast<::std::uint8_t>(StatusSubCode::kMaxSubCode) == 17, "disagrees with the value in #[cxx::bridge]");
warning:       |               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~
warning: /home/v/workspace/cozo/target/debug/build/cozorocks-4c4da2415a34542d/out/cxxbridge/sources/cozorocks/src/bridge/mod.rs.cc:876:71: note: the comparison reduces to ‘(16 == 17)’
```

--- a/cozo-bin/src/server.rs
+++ b/cozo-bin/src/server.rs
@@ -278,7 +278,7 @@ pub(crate) async fn server_main(args: ServerArgs) {
         args.engine, addr
     );
 
-    let listener = TcpListener::bind("127.0.0.1:3000").await.unwrap();
+    let listener = TcpListener::bind(&addr).await.unwrap();
     axum::serve(listener, app.into_make_service()).await.unwrap();
 }
 

